<?xml version="1.0"?>
<launch>


    <!--Run rviz-->
    <node pkg = "rviz" name = "rviz" type = "rviz" args="-d $(find robotsim)/rviz/drone5.rviz">
    </node>


    <!--ROBOT 0-->

    <!--Run simulated vision-->
    <node pkg = "robotsim" name = "vision" type = "vision.py" output="screen">
        <param name="Robot_number" type="int" value="0" />
        <param name="Number_of_robots" type="int" value="5" />
    </node>

    <!--Run the node that will generate an example path-->
    <node pkg = "robotsim" name = "robot" type = "drone_sim.py" args="" output="screen">
        <param name="Robot_number" type="int" value="0" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find robotsim)/config/drone_sim.yaml" />
    </node>


    <!--Run the node that will generate an example path-->
    <node pkg = "examples" name = "path_generator" type = "example_path.py" args="" output="screen">
        <param name="Robot_number" type="int" value="0" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find examples)/config/pub_path.yaml" />
    </node>


    <!-- Run the node that controls the robot with vector fields -->
    <node pkg="quad_robot" name="quad_node" type="quad_node.py" args="" output="screen">
        <param name="Robot_number" type="int" value="0" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find quad_robot)/config/quad_params.yaml" />
    </node>
    

    <!--ROBOT 1-->

    <!--Run simulated vision-->
    <node pkg = "robotsim" name = "vision1" type = "vision.py" output="screen">
        <param name="Robot_number" type="int" value="1" />
        <param name="Number_of_robots" type="int" value="5" />
    </node>

    <!--Run the node that will generate an example path-->
    <node pkg = "robotsim" name = "robot1" type = "drone_sim.py" args="" output="screen">
        <param name="Robot_number" type="int" value="1" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find robotsim)/config/drone_sim.yaml" />
    </node>


    <!--Run the node that will generate an example path-->
    <node pkg = "examples" name = "path_generator1" type = "example_path.py" args="" output="screen">
        <param name="Robot_number" type="int" value="1" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find examples)/config/pub_path.yaml" />
    </node>


    <!-- Run the node that controls the robot with vector fields -->
    <node pkg="quad_robot" name="quad_node1" type="quad_node.py" args="" output="screen">
        <param name="Robot_number" type="int" value="1" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find quad_robot)/config/quad_params.yaml" />
    </node>


    <!--ROBOT 2-->

    <!--Run simulated vision-->
    <node pkg = "robotsim" name = "vision2" type = "vision.py" output="screen">
        <param name="Robot_number" type="int" value="2" />
        <param name="Number_of_robots" type="int" value="5" />
    </node>

    <!--Run the node that will generate an example path-->
    <node pkg = "robotsim" name = "robot2" type = "drone_sim.py" args="" output="screen">
        <param name="Robot_number" type="int" value="2" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find robotsim)/config/drone_sim.yaml" />
    </node>


    <!--Run the node that will generate an example path-->
    <node pkg = "examples" name = "path_generator2" type = "example_path.py" args="" output="screen">
        <param name="Robot_number" type="int" value="2" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find examples)/config/pub_path.yaml" />
    </node>


    <!-- Run the node that controls the robot with vector fields -->
    <node pkg="quad_robot" name="quad_node2" type="quad_node.py" args="" output="screen">
        <param name="Robot_number" type="int" value="2" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find quad_robot)/config/quad_params.yaml" />
    </node>


    <!--ROBOT 3-->

    <!--Run simulated vision-->
    <node pkg = "robotsim" name = "vision3" type = "vision.py" output="screen">
        <param name="Robot_number" type="int" value="3" />
        <param name="Number_of_robots" type="int" value="5" />
    </node>

    <!--Run the node that will generate an example path-->
    <node pkg = "robotsim" name = "robot3" type = "drone_sim.py" args="" output="screen">
        <param name="Robot_number" type="int" value="3" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find robotsim)/config/drone_sim.yaml" />
    </node>


    <!--Run the node that will generate an example path-->
    <node pkg = "examples" name = "path_generator3" type = "example_path.py" args="" output="screen">
        <param name="Robot_number" type="int" value="3" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find examples)/config/pub_path.yaml" />
    </node>


    <!-- Run the node that controls the robot with vector fields -->
    <node pkg="quad_robot" name="quad_node3" type="quad_node.py" args="" output="screen">
        <param name="Robot_number" type="int" value="3" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find quad_robot)/config/quad_params.yaml" />
    </node>

    <!--ROBOT 4-->

    <!--Run simulated vision-->
    <node pkg = "robotsim" name = "vision4" type = "vision.py" output="screen">
        <param name="Robot_number" type="int" value="4" />
        <param name="Number_of_robots" type="int" value="5" />
    </node>

    <!--Run the node that will generate an example path-->
    <node pkg = "robotsim" name = "robot4" type = "drone_sim.py" args="" output="screen">
        <param name="Robot_number" type="int" value="4" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find robotsim)/config/drone_sim.yaml" />
    </node>


    <!--Run the node that will generate an example path-->
    <node pkg = "examples" name = "path_generator4" type = "example_path.py" args="" output="screen">
        <param name="Robot_number" type="int" value="4" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find examples)/config/pub_path.yaml" />
    </node>


    <!-- Run the node that controls the robot with vector fields -->
    <node pkg="quad_robot" name="quad_node4" type="quad_node.py" args="" output="screen">
        <param name="Robot_number" type="int" value="4" />
        <param name="Number_of_robots" type="int" value="5" />
        <rosparam command="load" file="$(find quad_robot)/config/quad_params.yaml" />
    </node>

</launch>


